pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -37,20 +37,30 @@ type CoreDataModel interface {
 	mcpclient.Updater
 }
 
+// Options stores the configurable attributes of a Control
+type Options struct {
+	DomainSuffix string
+}
+
 // Controller is a temporary storage for the changes received
 // via MCP server
 type Controller struct {
 	configStoreMu sync.RWMutex
 	// keys [type][namespace][name]
 	configStore              map[string]map[string]map[string]model.Config
 	descriptorsByMessageName map[string]model.ProtoSchema
+	options                  Options
+	eventHandlers            map[string]func(model.Config, model.Event)
 }
 
 // NewController provides a new CoreDataModel controller
-func NewController() CoreDataModel {
+func NewController(options Options) CoreDataModel {
 	descriptorsByMessageName := make(map[string]model.ProtoSchema, len(model.IstioConfigTypes))
 	for _, descriptor := range model.IstioConfigTypes {
-		descriptorsByMessageName[descriptor.MessageName] = descriptor
+		// don't register duplicate descriptors for the same message name, e.g. auth policy
+		if _, ok := descriptorsByMessageName[descriptor.MessageName]; !ok {
```

Comment:
Can we merge both `AuthenticationPolicy` and `AuthenticationMeshPolicy` into a single policy? they both follow same properties including same validation, except `ClusterScoped`. Can we have one policy which allows for cluster-scoped true or false? The reason asking is, it seems a little odd to have two descriptors with the same message name. Also, it helps with the logic in apply  where we adjust the type name for mesh-scoped resources.

Reply:
I think long term we might want to merge the two in the Pilot API. The k8s API (CRD) would still have namespace and cluster-scoped versions for the purposes of RBAC. Galley would be responsible for normalizing (see https://github.com/istio/istio/issues/8469). 

IMO, that work is outside the scope of this PR. There might be other factors to consider as well, e.g. workload vs. service identity and how we might handle hierarchical scope of policy more generally.

Reply:
That seems totally reasonable, I was thinking long term too. I realized how much worked it involved after the comment :)